src/main/java/TaskList.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Manages a list of tasks for the chatbot.
+ * Supports adding tasks of different types (todo, deadline, event),
+ * marking tasks as done or undone, and displaying all tasks.
+ */
+public class TaskList {
+    public final Task [] taskList= new Task[100]; // Array to store tasks
+    public int taskNo = 0; // Counter for the number of tasks
+
+    /**
+     * Adds a task to the task list based on user input.
+     * The task can be of types: todo, deadline, event, or a general task.
+     * Parses the user input to determine the task type and details.
+     *
+     * @param userInput The full command entered by the user to add a task.
+     */
+    public void addTask(String userInput){
+        switch (userInput.split(" ")[0]){
+        case "todo":
+            taskList[taskNo] = new Todo(userInput.substring(5),taskNo+1);
+            break;
+        case "deadline":
+            String name = userInput.split(" /")[0].substring(9);
+            String by = userInput.split(" /")[1].substring(3);
+            taskList[taskNo] = new Deadline(name, taskNo+1, by);
+            break;
+        case "event":
+            name = userInput.split(" /")[0].substring(6);
+            String from = userInput.split(" /")[1].substring(5);
+            String to = userInput.split(" /")[2].substring(3);
+            taskList[taskNo]=new Event(name,taskNo+1, from, to);
+            break;
+        default:
+            taskList[taskNo]=new Task(userInput,taskNo+1);
+            break;
+        }
+        taskAddMessage();
+    }
+
+    /**
+     * Displays a message confirming the addition of a task.
+     * Also shows the current number of tasks in the list.
+     */
+    private void taskAddMessage() {
+        System.out.println("    " + "--------------");
+        System.out.println("    Got it. I've added this task:");
+        System.out.print("      ");
+        taskList[taskNo].printTask();
+        taskNo++;
+        System.out.println("    Now you have " + taskNo + " tasks in the list.");
+        System.out.println("    " + "--------------");
+    }
+
+    /**
+     * Marks a task as completed based on its position in the task list.
+     * If the task does not exist, prints an error message.
+     *
+     * @param n The index of the task in the task list (0-based).
+     */
+    public void markTask (int n){
+        if (n >= taskNo){
+            print("Sorry, task unfound.");
+        }else{
```

Comment:
```suggestion
        } else{
```

Perhaps a space could be left between the } and else{ . This is for code consistency. I notice this in other parts of your code as well